core/src/main/scala/iliad/SizeOf.scala

```diff
@@ -0,0 +1,13 @@
+package iliad
+
+import simulacrum.typeclass
+
+@typeclass
+trait SizeOf[A] {
+  def byteSize: Int
+}
+
+@typeclass
+trait VertexAttribTypeConversion[A] {
+  def baseType: iliad.kernel.VertexAttribType
+}
```

Comment:
I disagree.  There are cases where we use `SizeOf`, where we don't imply that something is a `VertexAttribType`, eg [drawElements](https://github.com/zainab-ali/iliad-1/blob/free/core/src/main/scala/iliad/gl/opengl.scala#L169)